website/static/js/nodesPrivacy.js

```diff
@@ -0,0 +1,280 @@
+        }[self.page()];
+    });
+
+    self.message = ko.computed(function() {
+        return {
+            warning: MESSAGES.makeProjectPublicWarning,
+            select: MESSAGES.selectNodes,
+            confirm: MESSAGES.confirmWarning
+        }[self.page()];
+    });
+
+    self.selectProjects =  function() {
+        self.page(self.SELECT);
+    };
+
+    self.confirmWarning =  function() {
+        var nodesState = ko.toJS(self.nodesState);
+        for (var node in nodesState) {
+            if (nodesState[node].changed) {
+                if (nodesState[node].public) {
+                    self.nodesChangedPublic().push(nodesState[node].title);
+                }
+                else {
+                    self.nodesChangedPrivate().push(nodesState[node].title);
+                }
+            }
+        }
+        self.page(self.CONFIRM);
+    };
+
+    self.confirmChanges =  function() {
+        var nodesState = ko.toJS(self.nodesState());
+        //patchNodesPrivacy(nodesState);
+        nodesState = Object.keys(nodesState).map(function(key) {
+            return nodesState[key];
+        });
+        var nodesChanged = nodesState.filter(function(node) {
+            return node.changed;
+        });
+        patchNodesPrivacy(nodesChanged).then(function () {
+            window.location.reload();
+        });
+    };
+
+    self.clear = function() {
+        self.nodesChangedPublic([]);
+        self.nodesChangedPrivate([]);
+        self.page(self.WARNING);
+    };
+
+    self.selectAll = function() {
+        var nodesState = ko.toJS(self.nodesState());
+        for (var node in nodesState) {
+            if (nodesState[node].canWrite) {
+                nodesState[node].public = true;
+                nodesState[node].changed = nodesState[node].public !== self.nodesOriginal[node].public;
+            }
+        }
+        self.nodesState(nodesState);
+        m.redraw(true);
+    };
+
+    self.selectNone = function() {
+        var nodesState = ko.toJS(self.nodesState());
+        for (var node in nodesState) {
+            if (nodesState[node].canWrite) {
+                nodesState[node].public = false;
+                nodesState[node].changed = nodesState[node].public !== self.nodesOriginal[node].public;
+
+            }
+        }
+        self.nodesState(nodesState);
+        m.redraw(true);
+    };
+
+    self.back = function() {
+        var self = this;
+        self.nodesChangedPublic([]);
+        self.nodesChangedPrivate([]);
+        self.page(self.SELECT);
+    };
+
+};
+
+function NodesPrivacy (selector, parentNodePrivacy) {
+    var self = this;
+    self.selector = selector;
+    self.$element = $(self.selector);
+    self.viewModel = new NodesPrivacyViewModel(parentNodePrivacy);
+    self.viewModel.fetchNodeTree().done(function(response) {
+        new NodesPrivacyTreebeard('nodesPrivacyTreebeard', response, self.viewModel.nodesState, self.viewModel.nodesOriginal);
```

Comment:
The div ID should not be hardcoded (what if you want to use this is a test environment that doesn't have nodesPrivacyTreebeard in the HTML?). Pass this in as a parameter instead.